ertviz/controllers/observation_response_controller.py

```diff
@@ -0,0 +1,177 @@
+    realizations = _get_univariate_misfits_boxplots(
+        response.univariate_misfits_df(selected_realizations),
+        color=color,
+    )
+    ensemble_plot = ResponsePlotModel(
+        realizations,
+        [],
+        dict(
+            xaxis={"title": "Index"},
+            yaxis={"title": "Unit TODO"},
+            hovermode="closest",
+            uirevision=True,
+        ),
+    )
+    return ensemble_plot
+
+
+def observation_response_controller(parent, app):
+    @app.callback(
+        Output(parent.uuid("response-selector"), "options"),
+        [Input(parent.uuid("ensemble-selection-store"), "data")],
+    )
+    def _set_response_options(selected_ensembles):
+        # Should either return a union of all possible responses or the other thing which I cant think of...
+
+        if not selected_ensembles:
+            raise PreventUpdate
+        ensemble_id, _ = selected_ensembles.popitem()
+        ensemble = parent.ensembles.get(
+            ensemble_id,
+            EnsembleModel(ref_url=ensemble_id, project_id=parent.project_identifier),
+        )
+        parent.ensembles[ensemble_id] = ensemble
+        return [
+            {
+                "label": response,
+                "value": response,
+            }
+            for response in ensemble.responses
+        ]
+
+    @app.callback(
+        Output(parent.uuid("response-selector"), "value"),
+        [Input(parent.uuid("response-selector"), "options")],
+        [State(parent.uuid("response-selector"), "value")],
+    )
+    def _set_responses_value(available_options, previous_selected_response):
+        if available_options and previous_selected_response in [
+            opt["value"] for opt in available_options
+        ]:
+            return previous_selected_response
+        if available_options and not previous_selected_response:
+            return available_options[0]["value"]
+        return ""
+
+    @app.callback(
+        Output(
+            {"id": parent.uuid("response-graphic"), "type": parent.uuid("graph")},
+            "figure",
+        ),
+        [
+            Input(parent.uuid("response-selector"), "value"),
+            Input(parent.uuid("yaxis-type"), "value"),
+            Input(parent.uuid("misfits-type"), "value"),
+        ],
+        [State(parent.uuid("ensemble-selection-store"), "data")],
+    )
+    def _update_graph(response, yaxis_type, misfits_type, selected_ensembles):
+        if response in [None, ""] or not selected_ensembles:
+            raise PreventUpdate
+
+        if misfits_type == "Summary":
+            data_dict = {}
+            colors = {}
+            for ensemble_id, color in selected_ensembles.items():
+                ensemble = parent.ensembles.get(ensemble_id, None)
+                data_dict[parent.ensembles[ensemble_id]._name] = ensemble.responses[
+                    response
+                ].summary_misfits_df(
+                    selection=None
+                )  # What about selections?
```

Comment:
comment you have left addressing a question to yourself? Is it solved ? should the comment go?

Reply:
That's comment for us for how to solve selection.